10.Mender-server/docs.md

```diff
@@ -7,6 +7,52 @@ shortcode-core:
 github: false
 ---
 
+## Mender 3.7.3
+
+_Released 03.21.2024_
+
```

Comment:
I think we need a DL for removing the statistics + tasks to follow-up in release tool and release checklist